apiv2/apiv2_utils.py

```diff
@@ -79,6 +81,22 @@ def initial(self, request, *args, **kwargs):
     def log_message(self, message):
         return log_message_helper(message, resource=self)
 
+    def finalize_response(self, request, response, *args, **kwargs):
+        """ This method is overriden to make a redirect when the user is using the interactive API browser and
+        with 'www' sub-domain. The problem is that we can't check if it's accessing through the interactive browser
+        inside the 'initial' method because it raises an exception when the user is not logged in, that exception is
+        handled by 'finalize_response' method of APIView.
+        """
+        response = super(GenericAPIView, self).finalize_response(request, response, *args, **kwargs)
+
+        # If the user is using the interactive API browser and the www in the domain we redirect to nowww.
+        host = request.get_host()
+        if type(response.accepted_renderer) == type(BrowsableAPIRenderer()) and 'www' in host:
+            domain = "https://%s" % Site.objects.get_current().domain
```

Comment:
let's use the scheme from `request` here - if it's http then we can rely on haproxy to redirect us again to https

by using the request scheme we can test locally more easily